src/client.rs

```diff
@@ -52,115 +64,105 @@ impl ExecutionClient {
         auth_rpc: Uri,
         auth_rpc_jwt_secret: JwtSecret,
         timeout: u64,
-        metrics: Option<Arc<ClientMetrics>>,
         payload_source: PayloadSource,
-    ) -> Result<Self, ExecutionClientError> {
+    ) -> Result<Self, ClientError> {
         let auth_layer = AuthClientLayer::new(auth_rpc_jwt_secret);
         let auth_client = HttpClientBuilder::new()
             .set_http_middleware(tower::ServiceBuilder::new().layer(auth_layer))
             .request_timeout(Duration::from_millis(timeout))
             .build(auth_rpc.to_string())?;
 
         Ok(Self {
-            auth_client: Arc::new(auth_client),
+            auth_client,
             auth_rpc,
-            metrics,
             payload_source,
         })
     }
 
+    #[instrument(
```

Comment:
how does this record the status code and payload source in the response for metrics? I see a lot of the metrics code have been removed

Reply:
@avalonche metrics for the duration of each span will be created automatically using the span processor. If an instrumented function named 'foo' is called, we will get a histogram metric created with name 'foo_duration'. Histograms also contain a 'count' so we get all the same info we had before, but much cleaner.

Reply:
Oh I think I see what you're saying here. Yea that's correct, the response code is no longer included as a metric label. I just had a look about how we could achieve that in a nice way and came across [this issue](https://github.com/tokio-rs/tracing/issues/3190) in tracing. Without the ability to do that, keeping the response codes as labels would be pretty messy. I personally don't see a huge loss here since we still span status as a label, but if you feel strongly about it I'm sure I can implement a work around?